tests/generators/test_generators.py

```diff
@@ -151,3 +152,47 @@ def test_generator_structure(classname):
         "prompt" in inspect.signature(g.generate).parameters
     ), f"{classname}.generate() must accept parameter prompt"
     # generate("") w/ empty string doesn't fail, does return list
+
+
+TESTABLE_GENERATORS = [
+    classname
+    for classname in GENERATORS
+    if classname
+    not in [
+        "generators.function.Multiple",  # requires mock local function not implemented here
+        "generators.function.Single",  # requires mock local function not implemented here
```

Comment:
are these functions mock?

Reply:
A mock class could be created I would like to see that done in a `tests/geneartors/test_function.py`.  Really I would like to see this test extracted as a shared test run by test files created for each generator, to allow a class specific fixture to be injected to meet instantiation prerequisites down the line.